tests/core/data/test_excel_data_node.py

```diff
@@ -689,3 +693,170 @@ def test_read_write_after_modify_path(self):
         dn.write(read_data)
         for sheet, df in dn.read().items():
             assert np.array_equal(df.values, read_data[sheet].values)
+
+    def test_exposed_type_custom_class_after_modify_path(self):
+        path = os.path.join(pathlib.Path(__file__).parent.resolve(), "data_sample/example.xlsx")  # ["Sheet1", "Sheet2"]
+        new_path = os.path.join(
+            pathlib.Path(__file__).parent.resolve(), "data_sample/example_2.xlsx"
+        )  # ["Sheet1", "Sheet2", "Sheet3"]
+        dn = ExcelDataNode("foo", Scope.PIPELINE, properties={"default_path": path, "exposed_type": MyCustomObject1})
+        assert dn.exposed_type == MyCustomObject1
+        dn.read()
+        dn.path = new_path
+        dn.read()
```

Comment:
Please add some assertion to test that dn.read returned the expected value.
Plus I believe you don't need to test with a second path. Only one is enough.